test/src/test/java/org/corfudb/runtime/collections/vavr/HashArrayMappedTrieTest.java

```diff
@@ -0,0 +1,183 @@
+package org.corfudb.runtime.collections.vavr;
+
+import io.vavr.Tuple2;
+import org.corfudb.runtime.ExampleSchemas.ExampleValue;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+@SuppressWarnings("checkstyle:magicnumber")
+public class HashArrayMappedTrieTest {
```

Comment:
In terms of testing, I recommend having groups of test scenarios for each of the node types extending AbstractNode. Tests cases can be systematically developed, and simplify reasoning about the different cases that each node needs to handle.